### PR TITLE
[Telink] build version info log prints

### PR DESCRIPTION
**Pull request by interfer**
Added prints of git info for Matter, Zephyr and HAL repos for all Telink apps.
This detailed info in the logs is made for more clear version handling across the teams.

Device will have it printed in the COM port like this:

> I: SW Version: 1, prerelease
I: Matter revision: 
I:       board: tlsr9518adk80d
I:       branch: master 7e0634cf-dirty 2024-10-30
I:       remote: https://github.com/project-chip/connectedhomeip.git
I:       build timestamp: 2024-11-05 17:44:24
I: Zephyr revision: 
I:       branch: develop 6392406e 2024-10-04
I:       remote: git@github.com:telink-semi/zephyr.git
I:       HAL commit: e5a21d1f 2024-09-23

Note: `build timestamp` displays the time when the binary was created.
`dirty` means some local changes are present.

And during build there're similar format logs from Cmake:

> Loading Zephyr default modules (Zephyr base (cached)).
-- Matter revision:
--       board: tlsr9518adk80d
--       branch: build-version-info 9de0af072280e0aed80a2858c0ce578f1189c235 2024-11-05
--       remote: https://github.com/project-chip/connectedhomeip.git
--       build timestamp: 2024-11-05 18:40:19
-- Zephyr revision:
--       branch: develop 6392406e414697b3064a1193ea37fde9bdc7966a 2024-10-04
--       remote: git@github.com:telink-semi/zephyr.git
-- HAL revision:
--       commit: e5a21d1fb577c339ca149db91a67764e00338c61 2024-09-23
-- Configuring done (3.3s)

### CODE REVIEW

File: examples/platform/telink/build_info.cmake [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+# Matter git info
+execute_process(
+    COMMAND git rev-parse HEAD
```
Comment (andy31415) [inline examples/platform/telink/build_info.cmake:18]: what is the timing of these commands? 

We have added some `place SHA of the repo during build` logic and the end result was that the build was slowed down significantly. I would rather we did not run git commands during build time.

Comment (andy31415) [reply examples/platform/telink/build_info.cmake:18]: Maybe make it a non-default build option. If I build things for my local validation of builds, I do not want to spend extra time for the build.

Comment (interfer) [reply examples/platform/telink/build_info.cmake:18]: @andy31415 all it took for commands from `build_info.cmake` is 1sec

> -- Version info get start time: 1730879694
-- End Time: 1730879695
-- Execution Time: 1 seconds

I agree with your point to put all of this under some condition/option, will discuss with team's needs.

**Review by s07641069 — APPROVED** [review]

File: examples/platform/telink/build_info.cmake (hunk 1 above)
Comment (interfer) [reply examples/platform/telink/build_info.cmake:18]: Made present only for `CONFIG_CHIP_APP_LOG_LEVEL > 3` (debug and verbose)

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
@interfer please try to avoid force pushing the branch so we can see deltas between one change to another.

**Comment by interfer** [comment]
@andy31415 No problem, I've just rebased the branch a few times (no changes) to keep it updated after Cirque job in `master` has been fixed.

**Pull request merged**